components/App.jsx

```diff
@@ -22,23 +22,25 @@ const App = () => {
 
     const [isSpeechBoxActive, setIsSpeechBoxActive] = useState(false);
     const [speechBoxData, isSpeechBoxData] = useState(dialogueObj.intro)
+
+    let location = useLocation();
     
     // I think this wants to be refactored to some other name
     const [isNav, setIsNav] = useState(false);
 
     // does this do way too much?
     const spriteClickHandler = () => {
-        if (isInitialSpriteClick) {
+        if(location.pathname === "/" && isInitialSpriteClick){
             setSpriteClicked(true);
             // useEffect for animation change here too
             setTimeout(() => {
                 setIsSpeechBoxActive(true);
             }, 500);
             setIsInitialSpriteClick(false);
-        } else {
-            setIsNav(true);
         }
-
+        else {
```

Comment:
This function reads: 

"If they're at root and its their first time clicking the sprite, do some work with state regarding the sprite being clicked and the speechbox. Oh, also, start toggling the nav if I did that."

Those are two very different tasks. On top of that, I'm not getting a hint about either task from the function name. 

I would clean this up by giving it a more appropriate name. Or, perhaps you could wrap two different function into this spriteClickHandler: one that happens if `isInitialSpriteClick` and the other function which is executed if it isn't the initial sprite click.